### PR TITLE
Remove duplicated requests sending

### PR DESCRIPTION
**Pull request by Fifok**
According to #41:

I reimplemented debounce function to much simpler version.
I added debounced version of setSearchParams.

### CODE REVIEW

File: src/EpiContentUsage/Frontend/Lib/hooks/useFilteredTableData.ts [hunk 1]
```diff
@@ -79,6 +83,16 @@ export function useFilteredTableData<TableDataType>({
   const [searchParams, setSearchParams] = useSearchParams();
   const location = useLocation();
 
+  const debouncedSetSearchParams = useDebounce<
+    (prev: URLSearchParams) => URLSearchParamsInit
```
Comment (mgorzanski) [inline src/EpiContentUsage/Frontend/Lib/hooks/useFilteredTableData.ts:87]: Can't be just `(prev: URLSearchParams) => URLSearchParams`?

**Review by mgorzanski — CHANGES_REQUESTED** [review]
Unfortunately, I think there is a huge problem with the current approach. Because of how debounce works, currently only last call of the `debouncedSetSearchParams` function is executed, all of the rest is being rejected. I think that we want to only `debounce` the last step - setting the search params in the router. We need to keep track of the changes along the function calls. To see the problem, try sorting by a column, you will see that the only param that is being updated is `page`, because that's the last one in a queue (last `debouncedSetSearchParams` call), the rest are being ignored.

File: src/EpiContentUsage/Frontend/Lib/hooks/useFilteredTableData.ts (hunk 1 above)
Comment (Fifok) [reply src/EpiContentUsage/Frontend/Lib/hooks/useFilteredTableData.ts]: Yes. You're right. Done

**Review by mgorzanski — APPROVED** [review]
It looks like it works correctly :)

**Pull request merged**